src/runtime/animate/index.ts

```diff
@@ -21,8 +21,12 @@ export function flip(node: Element, { from, to }: { from: DOMRect; to: DOMRect }
 	const transform = style.transform === 'none' ? '' : style.transform;
 
 	const [ox, oy] = style.transformOrigin.split(' ').map(parseFloat);
-	const dx = (from.left + from.width * ox / to.width) - (to.left + ox);
-	const dy = (from.top + from.height * oy / to.height) - (to.top + oy);
+	const scale = {
+		x: to.width / node.clientWidth,
+		y: to.height / node.clientHeight
+	};
+	const dx = ((from.left / scale.x) + (from.width / scale.x) * ox / (to.width / scale.x)) - ((to.left / scale.x) + ox);
+	const dy = ((from.top / scale.y) + (from.height / scale.y) * oy / (to.height / scale.y)) - ((to.top / scale.y) + oy);
```

Comment:
```suggestion
	const dx = ((from.left - to.left) * node.clientWidth + from.width * ox) / to.width - ox;
	const dy = ((from.top - to.top) * node.clientHeight + from.height * oy) / to.height - oy;
```

How about something like this? Should be equivalent, and less than half as many multiplications/divisions.

Reply:
oh, nice @Conduitry !
I've added your revision to the REPL for test; https://svelte.dev/repl/ab91bfc09cf845d5aceaf42b092a7e3d?version=3.46.3 ... looks good.

❓ Should I click "commit suggestion" ?

---

One gripe I have (which is just personal) is that these calcs feel really obtuse as a reader without context. That goes for the original code and fixes, too. Some of it is soft of self-documenting (`from.left`, `to.width`, etc) but the `node.clientWidth` and `ox` parts do not feel clear on their purpose, or value. Which is partly why I began by using a `scale` object to sort-of explain the purpose of that calc _(as it's really -- and evidently due to being missed by Rich -- not obvious that this is important)_.

Any thoughts on the above paragraph?

Reply:
although I just remembered ... this is runtime code, not compiled, so efficiency is paramount, never mind :P